Sprint 14

AlbertKhaydarov opened the pull request:
На главном экране — две вкладки: «Трекеры» и «Статистика»
Свёрстан главный экран «Трекеры»:
Большой заголовок
В навигационном баре есть кнопка для добавления новых трекеров и кастомное окно даты DatePicker согласно дизайну
отображается список трекеров "привычка" на день даты и  не выполненные "нерегулярные"
реализована логика отметки трекера выполненным (для "нерегулярных" скрывается выполненный)
в будущую дату отметить нельзя (сообщение об ошибке)
если список трекеров для отображения пуст (в том числе — как результат поиска и/или выбора даты), то появляется заглушка
выполнен чек-лист по деталям реализации трекеров
Реализованы экраны создания трекера
Экран "категории" сделан на моковых данных (можно отметить, будут менять порядок)
"Расписание" работает, можно выбрать, будет передано в новый трекер
 Можно создать трекер каждого типа (без категории, выбора цвета и эмодзи)

AlbertKhaydarov committed:
prepare project

AlbertKhaydarov committed:
add UIColors+Extensions, TabBarController, LaunchScreen

AlbertKhaydarov committed:
add main screen Trackers

AlbertKhaydarov committed:
add Large title

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
Merge branch 'sprint_14' of github.com:AlbertKhaydarov/Tracker into sprint_14

# Conflicts:
#	Tracker/Trackers View/TrackersViewControllers.swift

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
add EventController, edit TrackersVC

AlbertKhaydarov committed:
add NewHabitViewController

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
add filteredChoosedByDatePickerDate

AlbertKhaydarov committed:
edit completed mark

AlbertKhaydarov committed:
edit completed mark

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
add searchController function

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
add setupCustomDatePickerView

AlbertKhaydarov committed:
add createdNewTracker

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit

AlbertKhaydarov committed:
edit sprint14 checklist

LizaIlinykh commented on Tracker/CreateNewTracker/EventTypeViewController.swift:
```diff
@@ -0,0 +1,90 @@
+        return button
+    }()
+    
+    private var viewRouter: RouterProtocol?
```
рекомендую вносить зависимости через инициализатор
вот статья про внедрение зависимостей https://swiftblog.org/vnedrenie-zavisimostej/

AlbertKhaydarov replied:
Выполнено

LizaIlinykh commented on Tracker/Trackers/TrackersViewControllers.swift:
```diff
@@ -0,0 +1,85 @@
+    
+}
+
+//extension TrackersViewController: UISearchResultsUpdating  {
```
лишний код лучше удалять

AlbertKhaydarov replied:
Выполнено

AlbertKhaydarov committed:
edit
установил минимальную версию - 13.4 (скорректировал функции, добавил class NewTrackerVieCell)
если несколько раз выбрать и убрать день недели, то дни отобразятся в списке несколько раз -  исправил (добавил сортировку)
у ячейки Расписание на экране создания трекера не должно быть нижнего сепаратора - исправил

AlbertKhaydarov committed:
на экране со списком трекеров в лейбле с количеством дней отображается только цифра, вместо "1 день" просто "1"
если нажать на кнопку плюс для трекера - нерегулярного события , он исчезает из коллекции
заглушка при поиске должна быть другой
для Router сдална зависимось через инициализатор
Tracker/Trackers/TrackersViewControllers.swift - удален лишний код

AlbertKhaydarov committed:
Елизавета, привет! спасибо большое за внимательное ревью, очень надеюсь, на этот раз должно все устроить!
Критические замечания:
если создаю привычку для воскресенья, она отображается по субботам (то есть происходит смещение трекера на день) - исправил + переделал сортировку
Рекомендации :
при вводе названия трекера в поисковой строке нужно также фильтровать трекеры и по дате - исправил
лучше закрывать экран со списком категорий после того как была выбрана Категория -  исправил + убрал лишний функционал
при переходе на экран с Расписанием , если ранее были выбраны дни, отобразить их выбранными - исправил + добавил UserDefaults

переделал таблицы для Расписания и Категорий для более корректного соответствия дизайну ( отступы и кастомные сепараторы)
